includes/OptionsHandler.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * WordPress Options-table Session Handler
+ *
+ * Like the original WP Session Handler, this class uses the WordPress
+ * Options table for data storage.
+ *
+ * @package WP Session Manager
+ * @subpackage Handlers
+ * @since 2.0
+ */
+namespace EAMann\Sessionz\Handlers;
+
+use EAMann\Sessionz\Handler;
+use EAMann\Sessionz\Objects\Option;
+
+/**
+ * Use an associative array to store session data so we can cut down on
+ * round trips to an external storage mechanism (or just leverage an in-
+ * memory cache for read performance).
+ */
+class OptionsHandler implements Handler {
+
+	/**
+	 * Sanitize a potential Session ID so we aren't fetching broken data
+	 * from the options table.
+	 *
+	 * @param string $id
+	 *
+	 * @return string
+	 */
+	private function sanitize( $id ) {
+		return preg_replace( "/[^A-Za-z0-9_]/", '', $id );
+	}
+
+	/**
+	 * Helper function to either add or update an existing option in the table.
+	 *
+	 * @param string $id
+	 * @param mixed  $data
+	 */
+	private function add_or_update( $id, $data ) {
+		if ( ! add_option( $id, $data, '', 'no' ) ) {
+			update_option( $id, $data );
+		}
+	}
+
+	/**
+	 * Get an item out of a WordPress option
+	 *
+	 * @param string $id
+	 *
+	 * @return bool|string
+	 */
+	protected function _read( $id ) {
+		$session_id = $this->sanitize( $id );
+
+		$data = get_option( "_wp_session_$session_id" );
+		$expires = intval( get_option( "_wp_session_expires_$session_id" ) );
+		if ( false !== $data ) {
+			/** @var Option $item */
+			$item = new Option( $data, $expires );
+			if ( ! $item->is_valid() ) {
+				delete_option( "_wp_session_$session_id" );
```

Comment:
The `delete()` method is part of a chain that does some other work _after_ deleting things. Could probably abstract the actual deletion to make the code more DRY. 👍